### ISSUE TITLE
Accessing HTML rendered elements from markdown

### ISSUE THREAD

**Issue by GerkinDev** [dimpu/ngx-md#140]
Hi,

I'm willing to access the HTML rendered inside an ngxMd element.

So far, I had the following code to get the HTML elements inside the ngxMd module:

```
import { NgxMdComponent } from 'ngx-md';

export class MarkdownContainerComponent implements OnInit {
	@ViewChild(NgxMdComponent) private markdownComponent?: NgxMdComponent;

	public ngOnInit() {
		const element = ((this.markdownComponent as any)._el as ElementRef).nativeElement;
		const childNodes = element.childNodes;
		console.log({ element, childNodes });
	}
}
```

But this code shows an empty list of child nodes. The component just not rendered yet. But how can I wait for the content to be accessed.

Moreover, the cast I did to bypass the `private` visibility of the NgxMdComponent is pretty annoying, even if I can imagine why it should be private.

So, what do you suggest?

Cheers
Gerkin

**Comment by dimpu** [comment]
Hi @GerkinDev 

It's a Viewchild so you need use ngViewInit Hook


```ts
import { Component, ViewChild } from '@angular/core';
import { NgxMdComponent } from 'ngx-md';

@Component({
  selector: 'my-app',
  templateUrl: './app.component.html',
  styleUrls: [ './app.component.css' ]
})
export class AppComponent  {
  @ViewChild(NgxMdComponent) main: any;

  name = 'NGX MD';
  constructor() {
    
  }

  ngAfterViewInit() {
    console.log(this.main);
    // for html
    console.log(this.main._md);
    // for nativeElement
    console.log(this.main._el.nativeElement)
  }
}

```

Hope this is helpful

**Comment by GerkinDev** [comment]
Hi,

Yeah, that's what I've deduced. But there's still something I want to talk about:

The code you wrote use `any` type instead of `NgxMdComponent`, thus, we are bypassing visibility of properties. This is not a good point, because you know what is the type of `main`. A stricter OOP language would totally disallow that. So, why the view is not exposed? Isn't it possible to make some accessors that I could use instead of bypassing properties protection?

Thanks!

**Comment by dimpu** [comment]
resolved with 6.0.9
